source-manager/src/main/java/com/jointrivial/sourcemanager/nordigen/web/controller/NordigenController.java

```diff
@@ -59,15 +71,31 @@ public NordigenController(CountryService countryService, BankService bankService
     public ResponseEntity<AuthorizationLinkViewModel> getAuthorizationLink(
             @RequestHeader("User-Token") String userToken,
             @RequestHeader("Bank-Id") String bankId,
+            @RequestHeader("Bank-Name") String bankName,
             @RequestHeader("Redirect-Link-Prefix") String redirectLinkPrefix) throws IOException, InterruptedException {
 
         // TODO check if userToken is valid with a request to account microservice
-
         AuthorizationLinkViewModel authorizationLinkViewModel =
-                this.sourceIdentifierService.getAuthorizationLink(userToken, bankId, redirectLinkPrefix);
+                this.nordigenConnectionIdService.getAuthorizationLink(userToken, bankId, bankName, redirectLinkPrefix);
 
         // TODO add exc handler
 
         return new ResponseEntity<>(authorizationLinkViewModel, HttpStatus.OK);
     }
+
+    @GetMapping(value = "/verify_requisition",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    public HttpStatus verifyRequisition(
+            @RequestHeader("User-Token") String userToken,
+            @RequestHeader("Reference-Id") String referenceId) throws IOException, InterruptedException {
+
+        NordigenConnectionIdServiceModel currentSourceIdentifier = this.nordigenConnectionIdService.getSourceIdentifierByReferenceId(referenceId);
+
+        String requisitionById = this.nordigen.getRequisitionById(currentSourceIdentifier.getRequisitionId());
```

Comment:
Calling the Nordigen's API and parsing it from Json logic makes more sense to be in the service layer. There, you can get what you need from the repo directly.